resources/type-system/inference.md

```diff
@@ -1117,6 +1134,26 @@ succintly, the syntax of Dart is extended to allow the following forms:
   any loss of functionality, provided they are not trying to construct a proof
   of soundness._
 
+In addition, the following forms are added to allow constructor invocations,
+dynamic method invocations, function object invocations, instance method
```

Comment:
I couldn't parse out of here which forms are intended to cover which subsets of the semantic space.  I think it would be helpful to be clear about that, even if it's not essential to your point.  For example, does `dynamic e; e.foo()` correspond to a DYNAMIC_INVOKE of the call method?  Or a FUNCTION_INVOKE?

Reply:
Added some informative text to clarify. Let me know if this section makes more sense now.